File: core/src/main/java/net/rickiekarp/core/controller/LanguageController.java
package net.rickiekarp.core.controller;

import kotlin.jvm.JvmStatic;
import kotlin.jvm.internal.Intrinsics;
import kotlin.text.StringsKt;
import net.rickiekarp.core.debug.LogFileHandler;
import net.rickiekarp.core.settings.Configuration;

import java.io.IOException;
import java.io.InputStream;
import java.util.Locale;
import java.util.Properties;

public final class LanguageController {
    private static Properties prop;

    public static final int getCurrentLocale() {
        String var0 = Configuration.CURRENT_LOCALE.toString();
        switch(var0.hashCode()) {
            case 3201:
                if (var0.equals("de")) {
                    return 1;
                }
                break;
            case 3241:
                if (var0.equals("en")) {
                    return 0;
                }
        }

        return 0;
    }

    public static final int getLocale() {
        String var10000 = System.getProperty("user.language");
        if (var10000 != null) {
            String var0 = var10000;
            switch(var0.hashCode()) {
                case 3201:
                    if (var0.equals("de")) {
                        return 1;
                    }
                    break;
                case 3241:
                    if (var0.equals("en")) {
                        return 0;
                    }
            }
        }

        return 0;
    }

    @JvmStatic
    public static final void loadLangFile(InputStream utf8in) {
        try {
            System.out.println(utf8in);
            System.out.println("Locale: " + Configuration.CURRENT_LOCALE);
            try {
                prop.load(utf8in);
            } catch (IOException var2) {
                var2.printStackTrace();
            }

            utf8in.close();
        } catch (NullPointerException | IOException var3) {
            LogFileHandler.logger.warning("Error");
        }

    }

    @JvmStatic
    public static final String getString(String textID) {
        Intrinsics.checkParameterIsNotNull(textID, "textID");

        try {
            String value = prop.getProperty(textID);
            CharSequence var2 = (CharSequence)value;
            return var2 == null || StringsKt.isBlank(var2) ? textID : value;
        } catch (NullPointerException var3) {
            LogFileHandler.logger.warning("Error when loading text ID: " + textID);
            return textID;
        }
    }

    @JvmStatic
    public static final void setCurrentLocale() {
        switch(Configuration.language) {
            case 0:
                Configuration.CURRENT_LOCALE = Locale.ENGLISH;
                break;
            case 1:
                Configuration.CURRENT_LOCALE = Locale.GERMAN;
                break;
            default:
                Configuration.CURRENT_LOCALE = Locale.ENGLISH;
        }

    }

    static {
        prop = new Properties();
    }
}

File: core/src/main/java/net/rickiekarp/core/controller/LanguageController.kt
package net.rickiekarp.core.controller


import net.rickiekarp.core.debug.LogFileHandler
import net.rickiekarp.core.settings.Configuration
import java.io.IOException
import java.util.Locale
import java.util.Properties

/**
 * Language controller class.
 */
object LanguageController {

    private var prop: Properties

    init {
        prop = Properties()
    }

    /**
     * Checks which language is used on the system and sets program specific language variables respectively.
     */
    @JvmStatic
    val currentLocale: Int
        get() {
            when (Configuration.CURRENT_LOCALE.toString()) {
                "en" -> return 0
                "de" -> return 1
                else -> return 0
            }
        }

    /**
     * Checks which language is used on the system and returns the respective integer.
     */
    @JvmStatic
    val locale: Int
        get() {
            when (System.getProperty("user.language")) {
                "en" -> return 0
                "de" -> return 1
                else -> return 0
            }
        }

    /**
     * Loads the language .properties file
     */
    @JvmStatic
    fun loadLangFile() {
        try {
            val utf8in = LanguageController::class.java.classLoader.getResourceAsStream("language_packs/language_" + Configuration.CURRENT_LOCALE + ".properties")

            try {
                prop.load(utf8in)
            } catch (e: IOException) {
                e.printStackTrace()
            }

            utf8in.close()
        } catch (e1: NullPointerException) {
            LogFileHandler.logger.warning("Error")
            //            e1.printStackTrace();
        } catch (e1: IOException) {
            LogFileHandler.logger.warning("Error")
        }

    }

    /**
     * Returns the text string of the given TextID
     * If no text string can be found, the TextID is returned.
     */
    @JvmStatic
    fun getString(textID: String): String {
        try {
            // get the property value
            val value = prop.getProperty(textID)

            return if (value.isNullOrBlank()) {
                textID
            } else {
                value
            }
        } catch (e1: NullPointerException) {
            LogFileHandler.logger.warning("Error when loading text ID: $textID")
            return textID
        }

    }

    /**
     * Checks which language is used on the system and sets program specific language variables respectively.
     */
    @JvmStatic
    fun setCurrentLocale() {
        when (Configuration.language) {
            0 -> Configuration.CURRENT_LOCALE = Locale.ENGLISH
            1 -> Configuration.CURRENT_LOCALE = Locale.GERMAN
            else -> Configuration.CURRENT_LOCALE = Locale.ENGLISH
        }
    }
}
